utils/python/CIME/SystemTests/system_tests_common.py

```diff
@@ -59,16 +58,19 @@ def _init_locked_files(self, caseroot, expected):
             shutil.copy(os.path.join(caseroot,"env_run.xml"),
                         os.path.join(lockedfiles, "env_run.orig.xml"))
 
-    def _init_case_setup(self):
+    def _resetup_case(self, phase):
         """
-        Do initial case setup needed in __init__
+        Re-setup this case. This is necessary if user is re-running an already-run
+        phase.
         """
-        if self._case.get_value("IS_FIRST_RUN"):
+        # We never want to re-setup if we're doing the resubmitted run
+        phase_status = self._test_status.get_status(phase)
+        if self._case.get_value("IS_FIRST_RUN") and phase_status != TEST_PEND_STATUS:
+
+            logging.warning("Resetting case due to detected re-run of phase %s" % phase)
             self._case.set_initial_test_values()
 
-        case_setup(self._case, reset=True, test_mode=True)
-        self._case.set_value("TEST",True)
-        self._case.flush()
+            case_setup(self._case, reset=True, test_mode=True, no_status=True)
```

Comment:
By setting no_status=True here, what will happen if this case_setup raises an error? I mainly ask to make sure that any error generated here will still result in a FAIL somewhere in TestStatus....